newrelic_api/muting_rules.py

```diff
@@ -0,0 +1,417 @@
+            mutation CreateMutingRule(
+                $accountId: Int!,
+                $ruleName: String!,
+                $ruleDescription: String!,
+                $ruleEnabled: Boolean!,
+                $ruleCondition: AlertsMutingRuleConditionGroupInput!,
+                $ruleSchedule: AlertsMutingRuleScheduleInput!,
+                $scheduleEnabled: Boolean!
+            ) {
+                alertsMutingRuleCreate(
+                    accountId: $accountId,
+                    rule: {
+                        name: $ruleName,
+                        description: $ruleDescription,
+                        enabled: $ruleEnabled,
+                        condition: $ruleCondition,
+                        schedule: $ruleSchedule
+                }) {
+                    id
+                    schedule @include(if: $scheduleEnabled) {
+                        endTime
+                        startTime
+                        timeZone
+                        repeat
+                        repeatCount
+                        endRepeat
+                        weeklyRepeatDays
+                        nextStartTime
+                        nextEndTime
+                    }
+                }
+            }"""
+
+        variables = {
+            "accountId": account_id,
+            "ruleName": rule_name,
+            "ruleDescription": rule_description,
+            "ruleEnabled": rule_enabled,
+            "ruleCondition": rule_condition,
+            "ruleSchedule": rule_schedule,
+            "scheduleEnabled": schedule_enabled,
+        }
+
+        return self._post(
+            self.URL,
+            headers=self.json_headers(),
+            data={"query": query, "variables": variables},
+        )
+
+    def delete_muting_rule(self, account_id, rule_id):
+        """
+        This API endpoint allows you to delete a muting rule
+
+        :type account_id: string
+        :param account_id: The muting rule's account ID.
+
+        :type rule_id: string
+        :param rule_id: The unique identifier for the muting rule.
+
+        :rtype: dict
+        :return: The JSON response of the API
+
+        ::
+            {
+                'data': {
+                    'alertsMutingRuleDelete': {
+                        'id': 'integer'
+                    }
+                }
+            }
+        """
+
+        query = """
+            mutation DeleteMutingRule(
+                $accountId: Int!,
+                $ruleId: ID!
+            ) {
+                alertsMutingRuleDelete(accountId: $accountId, id: $ruleId) {
+                    id
+                }
+            }
+        """
+
+        variables = {
+            "accountId": account_id,
+            "ruleId": rule_id,
+        }
+
+        return self._post(
+            self.URL,
+            headers=self.json_headers(),
+            data={"query": query, "variables": variables},
+        )
+
+    def list_muting_rule(self, account_id, rule_id):
```

Comment:
Would 'get_muting_rule' make more sense as a function name?

Reply:
Yes you are correct. It would be more sane, as currently it is too close to list_muting_rules, which does only get a list, while list_muting_rule gets the rules details.

Reply:
Updated